### ISSUE TITLE
Issue extracting files using Windows.KapeFiles.Extract

### ISSUE THREAD

**Issue by mgdevsoft** [Velocidex/velociraptor#2325]
Hi,
When using the Windows.KapeFiles.Extract artifact in order to preserve the timestamps the files inside the zip cannot be found due to the URL encoding inside the zip container.
So, basically:
When you open the zip file the path you have is:
%5C%5C.%5Cc%3A\$MFT
and when trying to extract the file the artifact is trying to access:
\\\\.\\c:\\$MFT   
and you get file not found when it needs to be created in the output folder.
just to emphasize not a single file is being created in the output directory.

The logic behind is to create a pipe line for log2timeline and timesketch.
For reference I looked at the project  in:
https://github.com/ReconInfoSec/velociraptor-to-timesketch

and there they just extract the zip files so timestamps are not preserved.
This can lead to distortion in the data in the analysis stage where timestamps have a key role.

Some information regarding my setup:
velociraptor server is running on Linux (ubuntu)
velociraptor client is running on Windows 10
Analysis workstation is Windows 10 (not the same as the running client)

Steps taken:
Running Windows.KapeFiles.Targets with do not be lazy and Kape triage .
Downloading the zip file to the analysis workstation.
trying to extract the files using Windows.KapeFiles.Extract via the GUI and via command line (on the workstation)
Results are the same.
all the argument where changed to fit the zip file structure.
for example:
ContainerPath: c:\Users\user\Downloads\hostname-clientid-flowid.zip
OutputDirectory: c:\output_dir
MetadataFile: client\host\artifacts\Windows.KapeFiles.Targets\flowid\All File Metadata.json
UploadsFile: client\host\artifacts\Windows.KapeFiles.Targets\flowid\Uploads.json

Partial Json from the debug info:
{
  "ZipPath": "\\\\.\\C:\\$MFT",
  "Created": "2017-02-03T03:16:54.0947633Z",
  "LastAccessed": "2017-02-03T03:16:54.0947633Z",
  "Modified": "2017-02-03T03:16:54.0947633Z",
  "CreatedFile": {
   "Path": "",
   "Size": 0,
   "Error": "Not found."
  }
Same behavior on Linux system as an analysis workstation where the zip file reside.

I think the best solution will be either using a VHD/VHDX as a container for the files like the original Kape tool is using or fixing the artifact.

Kind Regards,
Gal Miller

**Comment by scudette** [comment]
What version are you using? This was reworked in 0.6.7

**Comment by mgdevsoft** [comment]
Yes, the last version 0.6.7-4.
But this issue was there for as far as I can remember.

**Comment by mgdevsoft** [comment]
Yes, it sounds like https://github.com/Velocidex/velociraptor/issues/2273.
You are correct fixed in 0.6.7-4
I was running with the 0.6.7-rc3 
you can close it
